core/src/test/java/mediumtest/server/events/ServerSentEventsMediumTests.java

```diff
@@ -0,0 +1,556 @@
+        .withServerSentEventsEnabled()
+        .withSonarQubeConnection("connectionId", sonarServerMock.baseUrl())
+        .withUnboundConfigScope("configScope")
+        .build();
+
+      bind("configScope", "connectionId", "projectKey");
+
+      await().atMost(Duration.ofSeconds(2))
+        .untilAsserted(() -> assertThat(requestedPaths()).containsExactly("/api/push/sonarlint_events?projectKeys=projectKey&languages=java,js"));
+    }
+
+    @Test
+    void should_not_subscribe_for_events_if_sonarcloud_connection() {
+      System.setProperty("sonarlint.internal.sonarcloud.url", sonarServerMock.baseUrl());
+      backend = newBackend()
+        .withEnabledLanguageInStandaloneMode(Language.JS)
+        .withExtraEnabledLanguagesInConnectedMode(Language.JAVA)
+        .withServerSentEventsEnabled()
+        .withSonarCloudConnection("connectionId")
+        .withUnboundConfigScope("configScope")
+        .build();
+
+      bind("configScope", "connectionId", "projectKey");
+
+      await().pollDelay(Duration.ofMillis(200)).atMost(Duration.ofSeconds(2))
+        .untilAsserted(() -> assertThat(requestedPaths()).isEmpty());
+    }
+
+    private void bind(String configScopeId, String connectionId, String projectKey) {
+      backend.getConfigurationService().didUpdateBinding(new DidUpdateBindingParams(configScopeId, new BindingConfigurationDto(connectionId, projectKey, true)));
+    }
+
+    @Test
+    void should_not_resubscribe_for_events_if_sonarqube_connection_and_binding_is_the_same() {
+      backend = newBackend()
+        .withEnabledLanguageInStandaloneMode(Language.JS)
+        .withExtraEnabledLanguagesInConnectedMode(Language.JAVA)
+        .withServerSentEventsEnabled()
+        .withSonarQubeConnection("connectionId", sonarServerMock.baseUrl())
+        .withBoundConfigScope("configScope", "connectionId", "projectKey")
+        .build();
+      await().atMost(Duration.ofSeconds(2)).untilAsserted(() -> assertThat(requestedPaths()).hasSize(1));
+
+      bind("configScope", "connectionId", "projectKey");
+
+      await().pollDelay(Duration.ofMillis(200)).atMost(Duration.ofSeconds(2)).untilAsserted(() -> assertThat(requestedPaths()).hasSize(1));
+    }
+  }
+
+  @Nested
+  class WhenUnbindingScope {
+    @Test
+    void should_not_resubscribe_for_events_if_sonarqube_connection() {
+      backend = newBackend()
+        .withEnabledLanguageInStandaloneMode(Language.JS)
+        .withExtraEnabledLanguagesInConnectedMode(Language.JAVA)
+        .withServerSentEventsEnabled()
+        .withSonarQubeConnection("connectionId", sonarServerMock.baseUrl())
+        .withBoundConfigScope("configScope", "connectionId", "projectKey")
+        .build();
+      await().atMost(Duration.ofSeconds(2)).untilAsserted(() -> assertThat(requestedPaths()).hasSize(1));
+
+      unbind("configScope");
+
+      await().pollDelay(Duration.ofMillis(200)).atMost(Duration.ofSeconds(2)).untilAsserted(() -> assertThat(requestedPaths()).hasSize(1));
```

Comment:
Same, maybe we need a way to assert that all background processes/events have be executed...

Reply:
Yes this could be useful, but for that we would need a way to flush the event bus, and I don't think that's possible with the current implementation. Another reason to switch to something else?